LoveAndHiphop/LoveAndHiphop/DetailViewController.swift

```diff
@@ -32,50 +32,59 @@ class DetailViewController: UITableViewController {
+                self.photoScrollView.contentSize.width = self.photoScrollView.frame.width * CGFloat(i+2)
+                imageView.isUserInteractionEnabled = false
+                self.photoScrollView.addSubview(imageView)
+            }
+            }, failure: {
+                
+        })
+        
+        let gifmanager = SwiftyGifManager(memoryLimit:40)
+        let gif = UIImage(gifName: "loveme_or_not_gif.gif")
+        self.animatedImageView.setGifImage(gif, manager: gifmanager)
+        
+        self.nameLabel.text = userObject?.fullName
+        self.occupationLabel.text = userObject?.occupation
+        self.hiphopIdentityLabel.text = userObject?.hiphopIdentity
+        self.aboutLabel.text = userObject?.about
+        self.locationLabel.text = userObject?.location
+        self.emailIdLabel.text = userObject?.email
+        self.tableView.reloadData()
         
-//        self.photoScrollView.delegate = self
-//        let imageView = UIImageView()
-//        imageView.setImageWith((self.userObject?.profileImageUrl)!)
-//        imageView.contentMode = .scaleAspectFit
-//        imageView.frame = CGRect(x: 0, y: 0, width: self.photoScrollView.frame.width, height: self.photoScrollView.frame.height)
-//        self.photoScrollView.contentSize.width = self.photoScrollView.frame.width
-//        self.photoScrollView.addSubview(imageView)
-//
-//        self.loadOtherImages(success: { (imageUrlArray: [URL]) in
-//            for i in 0..<imageUrlArray.count {
-//                let imageView = UIImageView()
-//                imageView.setImageWith((imageUrlArray[i]))
-//                imageView.contentMode = .scaleAspectFit
-//                let xPosition = self.view.frame.width * CGFloat(i+1)
-//                imageView.frame = CGRect(x: xPosition, y: 8, width: self.photoScrollView.frame.width, height: self.photoScrollView.frame.height)
-//                self.photoScrollView.contentSize.width = self.photoScrollView.frame.width * CGFloat(i+2)
-//                imageView.isUserInteractionEnabled = false
-//                self.photoScrollView.addSubview(imageView)
-//            }
-//        }, failure: {
-//
-//        })
-//
-//        let gifmanager = SwiftyGifManager(memoryLimit:40)
-//        let gif = UIImage(gifName: "loveme_or_not_gif.gif")
-//        self.animatedImageView.setGifImage(gif, manager: gifmanager)
-//
-//        self.nameLabel.text = userObject?.fullName
-//        self.occupationLabel.text = userObject?.occupation
-//        self.hiphopIdentityLabel.text = userObject?.hiphopIdentity
-//        self.aboutLabel.text = userObject?.about
-//        self.locationLabel.text = userObject?.location
-//        self.emailIdLabel.text = userObject?.email
-//        self.tableView.reloadData()
         
+        //TODO: The local array does not retain data after coming back from detailed view
```

Comment:
The code below checks if the array of likedByUsers for this selected user, has the objectId of the current user. In that case, we show that red heart (That the user has been liked by the current user). If not, we should the green heart (That the user can be liked by the current user). 

The icons can be changed for better User interaction.